lib/ae_mdw/activities.ex

```diff
@@ -182,41 +196,114 @@ defmodule AeMdw.Activities do
           end
 
         cursor =
-          case cursor do
-            {_height, txi_cursor, local_idx_cursor} -> {create_txi, txi_cursor, local_idx_cursor}
+          case txi_cursor do
             nil -> nil
+            txi_cursor when direction == :forward -> {create_txi, txi_cursor, @min_int}
+            txi_cursor when direction == :backward -> {create_txi, txi_cursor, @max_int}
           end
 
         state
         |> Collection.stream(Model.GrpIntContractCall, direction, key_boundary, cursor)
         |> Stream.map(fn {^create_txi, txi, local_idx} ->
-          Model.tx(block_index: {height, _mbi}) = State.fetch!(state, Model.Tx, txi)
-
-          {{height, txi, local_idx}, :int_contract_call}
+          {txi, {:int_contract_call, local_idx}}
         end)
 
       :not_found ->
         []
     end
   end
 
-  defp build_txs_stream(state, direction, account_pk, range, txi_cursor) do
+  defp build_txs_stream(state, direction, account_pk, txi_scope, txi_cursor) do
     state
-    |> Fields.account_fields_stream(account_pk, direction, range, txi_cursor)
-    |> Stream.transform({-1, -1, -1}, fn
-      {txi, tx_type, tx_field_pos}, {txi, height, local_idx} ->
-        {[{{height, txi, local_idx + 1}, {:field, tx_type, tx_field_pos}}],
-         {txi, height, local_idx + 1}}
+    |> Fields.account_fields_stream(account_pk, direction, txi_scope, txi_cursor)
+    |> Stream.map(fn {txi, tx_type, tx_field_pos} -> {txi, {:field, tx_type, tx_field_pos}} end)
+  end
+
+  defp build_aexn_transfers_stream(state, direction, account_pk, txi_scope, txi_cursor) do
+    transfers_stream =
+      state
+      |> build_aexn_transfer_stream(
+        Model.AexnTransfer,
+        direction,
+        account_pk,
+        txi_scope,
+        txi_cursor
+      )
+      |> Stream.map(fn {aexn_type, ^account_pk, txi, to_pk, value, index} ->
+        {txi, {:aexn, aexn_type, account_pk, to_pk, value, index}}
+      end)
+
+    rev_transfers_stream =
+      state
+      |> build_aexn_transfer_stream(
+        Model.RevAexnTransfer,
+        direction,
+        account_pk,
+        txi_scope,
+        txi_cursor
+      )
+      |> Stream.map(fn {aexn_type, ^account_pk, txi, from_pk, value, index} ->
+        {txi, {:aexn, aexn_type, from_pk, account_pk, value, index}}
+      end)
+
+    Collection.merge([transfers_stream, rev_transfers_stream], direction)
+  end
+
+  defp build_aexn_transfer_stream(state, table, direction, account_pk, txi_scope, txi_cursor) do
+    @aexn_types
+    |> Enum.map(fn aexn_type ->
+      key_boundary =
+        case txi_scope do
+          nil ->
+            {
+              {aexn_type, account_pk, @min_int, @min_int, @min_int, @min_int},
+              {aexn_type, account_pk, @max_int, @max_int, @max_int, @max_int}
```

Comment:
the fourth element is a pubkey, min and max bin would help with readability (or at least only min bin)

Reply:
Done